primitives/runtime/src/transaction_validity.rs

```diff
@@ -264,6 +264,17 @@ impl Default for ValidTransaction {
 }
 
 impl ValidTransaction {
+	/// Initiate `ValidTransaction` builder object with a particular prefix for tags.
+	///
+	/// To avoid conflicts between different pallets it's recommended to build `requires`
+	/// and `provides` tags with a unique prefix.
+	pub fn for_pallet(prefix: &'static str) -> ValidTransactionBuilder {
```

Comment:
I wouldn't call this for pallet but rather with_prefix. It is better to demonstrate exactly what is happening. 

also pallet is.. pallet and primitives should not be pallet oriented.

Reply:
> I wouldn't call this for pallet but rather with_prefix. It is better to demonstrate exactly what is happening.

I'd argue that it's less clear. What `prefix` is that? What does it apply to?

> also pallet is.. pallet and primitives should not be pallet oriented.

That's arguable as well. I feel that the `frame` and `sp-runtime` distinction is quite lost already many things are quite `frame` specific. Although it's a pretty lame argument, since it doesn't justify adding even more frame-specific things.

That said maybe `with_tags_prefix` would be a good common ground, wdyt?

Reply:
sounds good!